### ISSUE TITLE
Redirecting the keyboard to the serial port

### ISSUE THREAD

**Issue by toncho11** [eyalabraham/new-xt-bios#22]
Hi,

Is your BIOS IBM XT 5160 compatible?

I do not have a working keyboard, defective 74S74N and I would like to use the serial port to feed the keys instead of the keyboard. See below:

With the assumption that there is a serial port sitting at 3F8h, IRQ4

1.  POST within BIOS modified to initiate the serial port (set parameters, flow control, and enable IRQ) at computer power-on time.

2.  IRQ4 pointer modified to point to the keyboard handler routine.  IRQ1 pointer disabled.

3.  Keyboard handler routine (called for each keyboard byte received) modified per:
      - Fetch byte from the serial port at 3F8h, rather than keyboard shift register.
      - Clear IRQ4

Is this possible with your BIOS?

**Comment by eyalabraham** [comment]
@toncho11 unfortunately it is not 5160 compatible. I created the BIOS to fit the hardware that I built around my old motherboard and only made sure that DOS sees it as a valid BIOS for what it needed. By "sees it as valid" I mean that in the initial debug stages I literally traced DOS's usage (calls into my BIOS) and filled in the missing calls.

Also, my serial port is not standard in the sens that it is mapped to $300.

It is possible to do what you are looking for; feel free to use my code as a starting point if you wish.
